src/nlp/utils/logging.py

```diff
@@ -0,0 +1,191 @@
+    .. note::
+        HuggingFace Transformers has following logging levels:
+        - ``nlp.logging.CRITICAL``, ``nlp.logging.FATAL``
+        - ``nlp.logging.ERROR``
+        - ``nlp.logging.WARNING``, ``nlp.logging.WARN``
+        - ``nlp.logging.INFO``
+        - ``nlp.logging.DEBUG``
+    """
+
+    _configure_library_root_logger()
+    return _get_library_root_logger().getEffectiveLevel()
+
+
+def set_verbosity(verbosity: int) -> None:
+    """Set the level for the HuggingFace Transformers's root logger.
+    Args:
+        verbosity:
+            Logging level, e.g., ``nlp.logging.DEBUG`` and ``nlp.logging.INFO``.
+    """
+
+    _configure_library_root_logger()
+    _get_library_root_logger().setLevel(verbosity)
+
+
+def set_verbosity_info():
+    """Set the level for the HuggingFace Transformers's root logger to INFO.
+
+    This will display most of the logging information and tqdm bars.
+
+    Shortcut to ``nlp.logging.set_verbosity(nlp.logging.INFO)``
+    """
+    return set_verbosity(INFO)
+
+
+def set_verbosity_warning():
+    """Set the level for the HuggingFace Transformers's root logger to WARNING.
+
+    This will display only the warning and errors logging information (no tqdm bars).
+
+    Shortcut to ``nlp.logging.set_verbosity(nlp.logging.WARNING)``
+    """
+    return set_verbosity(WARNING)
+
+
+def set_verbosity_debug():
+    """Set the level for the HuggingFace Transformers's root logger to DEBUG.
+
+    This will display all the logging information and tqdm bars.
+
+    Shortcut to ``nlp.logging.set_verbosity(nlp.logging.DEBUG)``
+    """
+    return set_verbosity(DEBUG)
+
+
+def set_verbosity_error():
+    """Set the level for the HuggingFace Transformers's root logger to ERROR.
+
+    This will display only the errors logging information (no tqdm bars).
+
+    Shortcut to ``nlp.logging.set_verbosity(nlp.logging.ERROR)``
+    """
+    return set_verbosity(ERROR)
+
+
+def disable_default_handler() -> None:
+    """Disable the default handler of the HuggingFace Transformers's root logger."""
+
+    _configure_library_root_logger()
+
+    assert _default_handler is not None
+    _get_library_root_logger().removeHandler(_default_handler)
+
+
+def enable_default_handler() -> None:
+    """Enable the default handler of the HuggingFace Transformers's root logger."""
+
+    _configure_library_root_logger()
+
+    assert _default_handler is not None
+    _get_library_root_logger().addHandler(_default_handler)
+
+
+def disable_propagation() -> None:
+    """Disable propagation of the library log outputs.
+    Note that log propagation is disabled by default.
+    """
+
+    _configure_library_root_logger()
+    _get_library_root_logger().propagate = False
+
+
+def enable_propagation() -> None:
+    """Enable propagation of the library log outputs.
+    Please disable the HuggingFace Transformers's default handler to prevent double logging if the root logger has
```

Comment:
what would be the issue with transformers exactly ?

Reply:
copy-past error